lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,27 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't send unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.nil? || actions.size < 1
+      supported_actions, unsupported_actions = actions.partition { |action, _, _| LogStash::Outputs::ElasticSearch::VALID_HTTP_ACTIONS.include?(action) }
+      if !unsupported_actions.nil? && unsupported_actions.size > 0
+        handle_dlq_or_drop(unsupported_actions)
+        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: actions.size)
+      end
```

Comment:
- `Enumerable#partition` will _always_ return 2 arrays, so we don't need a nil-check here.
 - because we should be logging the actual events that are being dropped (or not logging when DLQ is enabled and can capture them), the additional warning here isn't particularly meaningful or actionable.

```suggestion
      if unsupported_actions.any?
        handle_dlq_or_drop(unsupported_actions)
        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: unsupported_actions.size)
      end
```

Or, since we log the skipped actions in the DLQ-or-drop handling:

```suggestion
      handle_dlq_or_drop(unsupported_actions) if unsupported_actions.any?
```

And if the `handle_or_drop(unsupported_actions)` method is implemented as-suggested by using `Enumerable#each`, we don't need to do the `Enumerable#any?` check here either.

```suggestion
      handle_dlq_or_drop(unsupported_actions)
```